### DISCUSSION TITLE
Autorun-rs v1.2.0

### DISCUSSION THREAD

**Discussion by vurvdev** [vurvdev/Autorun-rs#54] [Announcements]
> This update brings a file system, major optimizations to file size and a lot of changes you'd normally overlook to the lua api 🌑.

## Optimized File Size 📦
Autorun was getting pretty big from all of the dependencies.
Things were removed and replaced, ``human-panic`` was removed since it'd only show up for a few frames until gmod crashed since this is used as a dll, ``chrono`` replaced with ``time``, etc.

|   📦   | Old    | New    |
|--------|--------|--------|
| 32 Bit | 1 MB   | 570 KB |
| 64 Bit | 1.2 MB | 663 KB |

### Initial Freezing 🧊
Due to the ``autorun/settings.toml`` file not existing, your game would freeze. This was fixed.  
__Now Autorun will properly create all essential folders and the settings file for you__

### 32 Bit? 💻
As of this version, 32 bit seems to work properly.
Note it is missing the lua executor commands from the cli, which are still unable to be provided unfortunately.

## File System 🗄️
In order to accompany some new features here, I wrote a file system for Autorun that'll make the process of taking input paths from the user much easier, and adding functions like ``Autorun.readFile`` and ``Autorun.require`` much easier.

__This brings local paths to the lua api!__

## Plugin overrides 🧩
You can now return a string or boolean from a plugin's ``src/hook.lua``, just as you can from ``autorun/hook.lua``, to override the script!

### ``autorun.nocolor``
This setting is under the [autorun] tab as "nocolor", and disables the color in case [your system doesn't support truecolor](https://github.com/Vurv78/Autorun-rs/issues/50), or you simply don't want the colors.

### ``plugin``
This is a general command for managing and creating plugins with Autorun, subcommands below:
* ``new <name>`` - Creates a new plugin with template code at folder ``<folder>``
* ``list`` - Lists all currently (would-be) active plugins and info about them.
* ``help`` - Explains the other subcommands to you in greater detail

### ``Autorun.require`` 🌔
This function has been fixed to allow for calling it without having a .lua extension, bringing it closer to vanilla lua.

Additionally it now allows usage between local folders, so you can finally use it inside of your plugins 🧩!

Previously it only allowed you to ``Autorun.require`` files inside of your global ``autorun/scripts`` folder, but now you can require local to the running script.

__Example__
```lua
-- In autorun/plugins/myplugin/src/autorun.lua
local TheString = Autorun.require("myfile")
print(TheString) -- Hello world!
```

```lua
-- In autorun/plugins/myplugin/src/myfile.lua
return "Hello world!"
```

### ``Autorun.readFile`` 📂
The long awaited function to read a file is here!
Call it with a filename and read a file local to the currently running script, same as ``Autorun.require`` now can.

## Once Again...
That's pretty much it. It took a while to get here, so if you could support with a 🌟 or [join the discord to ask any questions or report issues](https://discord.gg/yXKMt2XUXm), it would be greatly appreciated. (This took way too long to write, so could support me for that as well.. 😵‍💫 )

<hr /><em>This discussion was created from the release <a href='https://github.com/Vurv78/Autorun-rs/releases/tag/v1.2.0'>Autorun-rs v1.2.0</a>.</em>

**Comment by Derpius** [comment]
Vurv stop writing readmes, issues, and releases like a front end web dev with an anime pfp :trollface:

**Reply by vurvdev** [reply]
No
